test/integration-tests/Element.test.ts

```diff
@@ -0,0 +1,67 @@
+import { CommandType, RoutePlanner } from './shared/planner'
+import ELEMENT_ABI from './shared/abis/Element.json'
+import { ERC721, UniversalRouter, Permit2 } from '../../typechain'
+import { resetFork } from './shared/mainnetForkHelpers'
+import { ALICE_ADDRESS, DEADLINE } from './shared/constants'
+import deployUniversalRouter, { deployPermit2 } from './shared/deployUniversalRouter'
+import { SignerWithAddress } from '@nomiclabs/hardhat-ethers/signers'
+import hre from 'hardhat'
+import { BigNumber } from 'ethers'
+import { abi as ERC721_ABI } from '../../artifacts/solmate/src/tokens/ERC721.sol/ERC721.json'
+import { expect } from 'chai'
+import { EXAMPLE_ETH_SELL_ORDER, EXAMPLE_ETH_SELL_ORDER_SIG } from './shared/protocolHelpers/element'
+
+const { ethers } = hre
+
+const ELEMENT_721_INTERFACE = new ethers.utils.Interface(ELEMENT_ABI)
+
+describe('Element Market', () => {
+  let alice: SignerWithAddress
+  let router: UniversalRouter
+  let permit2: Permit2
+  let planner: RoutePlanner
+  let testNFTContract: ERC721
+
+  const order = EXAMPLE_ETH_SELL_ORDER
+  const signature = EXAMPLE_ETH_SELL_ORDER_SIG
+  const nftContractAddress = order.nft
+
+  beforeEach(async () => {
+    // txn is at block 16627214
+    await resetFork(16627214 - 1)
```

Comment:
did you try to test on polygon itself? was it too hard?
I know infura are having polygon issues this week so not worth trying now haha

Reply:
I was running into a really weird bug where the polygon fork would only work if the default hardhat network is set to polygon - and when I did that polygon was working. But that would break all of our other tests